src/toil/provisioners/aws/awsProvisioner.py

```diff
@@ -126,28 +132,88 @@ def awsFilterImpairedNodes(nodes, ec2):
 class InvalidClusterStateException(Exception):
     pass
 
+class AWSConnectionManager:
+    """
+    Class that represents a connection to AWS. Caches Boto 3 and Boto 2 objects
+    by region.
+
+    Access to any kind of item goes through the particular method for the thing
+    you want (session, resource, service, Boto2 Context), and then you pass the
+    zone you want to work in, and possibly the type of thing you want, as arguments.
```

Comment:
I keyed this all by zone in the external API, but it turns out all AWS connections are scoped to region and not zone. Maybe we really need a `_region` on the provisioner so we can have it handy to use the API here?